core/templates/dev/head/pages/email_dashboard/EmailDashboardServices.js

```diff
@@ -53,66 +53,70 @@ oppia.factory('EmailDashboardDataService', [
         return latestCursor;
       },
 
-      submitQuery: function(data, callback) {
-        var startQueryIndex = (currentPageIndex) * QUERIES_PER_PAGE;
+      submitQuery: function(data) {
+        var startQueryIndex = currentPageIndex * QUERIES_PER_PAGE;
         var endQueryIndex = (currentPageIndex + 1) * QUERIES_PER_PAGE;
 
-        $http.post(QUERY_DATA_URL, {
+        return $http.post(QUERY_DATA_URL, {
           data: data
-        }).success(function() {
-          // Wait for 500ms and then fetch details of newly submitted query.
-          $timeout(function() {
-            fetchQueriesPage(1, null).then(function(data) {
-              var newQueries = [data.recent_queries[0]];
-              queries = newQueries.concat(queries);
-              callback(queries.slice(startQueryIndex, endQueryIndex));
-            });
-          }, 500);
+        }).then(function(response) {
+          var data = response.data;
```

Comment:
I really don't know which one would be better. I think we should let it be the way it is right now since it shows a message about query being submitted successfully after submitting it. We can change it later on if required.